src/mito2/src/cache/manifest_cache.rs

```diff
@@ -412,11 +427,16 @@ impl ManifestCache {
 
                 let subdir_empty = Self::remove_empty_dirs_recursive_sync(&path, check_mtime)?;
                 if subdir_empty {
-                    if let Err(e) = std::fs::remove_dir(&path)
-                        && e.kind() != std::io::ErrorKind::NotFound
-                    {
-                        warn!(e; "Failed to remove empty directory {}", path.display());
-                        is_empty = false;
+                    if let Err(e) = std::fs::remove_dir(&path) {
+                        if e.kind() != std::io::ErrorKind::NotFound {
+                            warn!(e; "Failed to remove empty directory {}", path.display());
+                            is_empty = false;
```

Comment:
Not found doesn't means it's not empty.

Reply:
remove_dir only removes empty dir. If it returns error and the error code isn't not found, then we don't know whether the dir is empty so we set  is_empty  to false. We only set is_empty to true when we ensure the dir is empty.